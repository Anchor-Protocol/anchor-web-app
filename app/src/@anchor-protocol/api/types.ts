export interface GovRewardsResponse {
  timestamp: number;
  height: number;
  current_apy: number;
}
