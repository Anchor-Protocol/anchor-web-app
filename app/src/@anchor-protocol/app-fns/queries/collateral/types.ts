import { moneyMarket } from '@anchor-protocol/types';
import { WasmQuery } from '@libs/query-client';

export interface WhitelistWasmQuery {
  whitelist: WasmQuery<
    moneyMarket.overseer.Whitelist,
    moneyMarket.overseer.WhitelistResponse
  >;
}

export type WhitelistCollateral =
  moneyMarket.overseer.WhitelistResponse['elems'][0] & {
    icon?: string;
    decimals?: number;
  };
