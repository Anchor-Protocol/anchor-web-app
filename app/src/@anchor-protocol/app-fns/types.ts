import {
  ANC,
  AncUstLP,
  aUST,
  bLuna,
  bLunaLunaLP,
  Luna,
  Eth,
  bEth,
  Native,
  Rate,
  u,
  UST,
  CollateralAmount,
} from '@anchor-protocol/types';
import Big from 'big.js';
import { WhitelistCollateral } from 'queries';

/**
 * You can cast the token values as nominal types
 *
 * @example
 * ```
 * // const { tokenBalances: { uUST } } = useBank() // Record<string, string>
 * const { tokenBalances: { uUST } } = useBank<AnchorTokens>() // { uUST: uUST }
 * ```
 */
export interface AnchorTokenBalances {
  uUST: u<UST>;
  uaUST: u<aUST>;
  uLuna: u<Luna>;
  ubLuna: u<bLuna>;
  uEth: u<Eth>;
  ubEth: u<bEth>;
  uANC: u<ANC>;
  uAncUstLP: u<AncUstLP>;
  ubLunaLunaLP: u<bLunaLunaLP>;
}

export const DefaultAnchorTokenBalances = {
  uUST: '0' as u<UST>,
  uaUST: '0' as u<aUST>,
  uLuna: '0' as u<Luna>,
  ubLuna: '0' as u<bLuna>,
  uEth: '0' as u<Eth>,
  ubEth: '0' as u<bEth>,
  uANC: '0' as u<ANC>,
  uAncUstLP: '0' as u<AncUstLP>,
  ubLunaLunaLP: '0' as u<bLunaLunaLP>,
};

export interface AnchorBalances {
  uNative: u<Native>; // the native token for the chain, ie, LUNA, ETH, AVAX
  uUST: u<UST>;
  uaUST: u<aUST>;
  uANC: u<ANC>;

  fetchWalletBalance: (
    collateral?: WhitelistCollateral,
  ) => Promise<u<CollateralAmount<Big>>>;
}

export const DefaultAnchorBalances: Partial<AnchorBalances> = {
  uNative: '0' as u<Native>,
  uUST: '0' as u<UST>,
  uaUST: '0' as u<aUST>,
  uANC: '0' as u<ANC>,
};

/**
 * You can cast the tax values as nominal types
 *
 * @example
 * ```
 * // const { tax: { taxRate, maxTaxUUSD } } = useBank() // { taxRate: string, maxTaxUUSD: string }
 * const { tax: { taxRate, maxTaxUUSD } } = useBank<any, AnchorTax>() // { taxRate: Rate, maxTaxUUSD: uUST }
 * ```
 */
export interface AnchorTax {
  taxRate: Rate;
  maxTaxUUSD: u<UST>;
}
