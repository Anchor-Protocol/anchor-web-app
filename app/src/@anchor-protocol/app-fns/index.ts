// THIS FILE IS AUTO CREATED
// @see ~/scripts/create-index.ts
export * from './env';
export * from './types';
export * from './caches/airdropStage';
export * from './forms/borrow/borrow';
export * from './forms/borrow/provideCollateral';
export * from './forms/borrow/redeemCollateral';
export * from './forms/borrow/repay';
export * from './forms/earn/deposit';
export * from './forms/earn/withdraw';
export * from './functions/cw20Decimals';
export * from './logics/borrow/computebAssetLtvsAvg';
export * from './logics/borrow/computeBorrowAmountToLtv';
export * from './logics/borrow/computeBorrowAPR';
export * from './logics/borrow/computeBorrowedAmount';
export * from './logics/borrow/computeBorrowLimit';
export * from './logics/borrow/computeBorrowMax';
export * from './logics/borrow/computeBorrowNextLtv';
export * from './logics/borrow/computeBorrowReceiveAmount';
export * from './logics/borrow/computeBorrowSafeMax';
export * from './logics/borrow/computeBorrowTxFee';
export * from './logics/borrow/computeCollateralsTotalUST';
export * from './logics/borrow/computeCurrentLtv';
export * from './logics/borrow/computeDepositAmountToBorrowLimit';
export * from './logics/borrow/computeDepositAmountToLtv';
export * from './logics/borrow/computeEstimateLiquidationPrice';
export * from './logics/borrow/computeLiquidationPrice';
export * from './logics/borrow/computeLtv';
export * from './logics/borrow/computeLtvToBorrowAmount';
export * from './logics/borrow/computeLtvToDepositAmount';
export * from './logics/borrow/computeLtvToRedeemAmount';
export * from './logics/borrow/computeLtvToRepayAmount';
export * from './logics/borrow/computeMaxRepayingAmount';
export * from './logics/borrow/computeNetAPR';
export * from './logics/borrow/computeProvideCollateralBorrowLimit';
export * from './logics/borrow/computeProvideCollateralNextLtv';
export * from './logics/borrow/computeRedeemAmountToLtv';
export * from './logics/borrow/computeRedeemCollateralBorrowLimit';
export * from './logics/borrow/computeRedeemCollateralNextLtv';
export * from './logics/borrow/computeRepayAmountToLtv';
export * from './logics/borrow/computeRepayNextLtv';
export * from './logics/borrow/computeRepaySendAmount';
export * from './logics/borrow/computeRepayTotalBorrowed';
export * from './logics/borrow/computeRepayTotalOutstandingLoan';
export * from './logics/borrow/computeRepayTxFee';
export * from './logics/borrow/pickCollateral';
export * from './logics/borrow/validateBorrowAmount';
export * from './logics/borrow/validateDepositAmount';
export * from './logics/borrow/validateRedeemAmount';
export * from './logics/borrow/validateRepayAmount';
export * from './logics/borrow/vectorizeBAssetLtvs';
export * from './logics/borrow/vectorizeOraclePrices';
export * from './logics/borrow/vectorizeOverseerCollaterals';
export * from './logics/borrow/vectorizeOverseerWhitelist';
export * from './logics/borrow/vectorizeVariations';
export * from './logics/common/validateTxFee';
export * from './logics/common/computeGasPrice';
export * from './logics/common/computeGasToUst';
export * from './logics/earn/computeCurrentAPY';
export * from './logics/earn/computeTotalDeposit';
export * from './models/collaterals';
export * from './queries/airdrop/check';
export * from './queries/airdrop/isClaimed';
export * from './queries/anc/balance';
export * from './queries/anc/lpStakingState';
export * from './queries/anc/price';
export * from './queries/anc/tokenInfo';
export * from './queries/basset/bAssetClaimableRewards';
export * from './queries/basset/bAssetClaimableRewardsTotal';
export * from './queries/basset/bAssetInfo';
export * from './queries/basset/bAssetInfoAndBalance';
export * from './queries/basset/bAssetInfoAndBalanceByTokenAddr';
export * from './queries/basset/bAssetInfoAndBalanceTotal';
export * from './queries/basset/bAssetInfoByTokenAddr';
export * from './queries/basset/bAssetInfoList';
export * from './queries/basset/bLunaClaimableRewards';
export * from './queries/basset/bLunaExchangeRate';
export * from './queries/basset/bLunaWithdrawableAmount';
export * from './queries/borrow/apy';
export * from './queries/borrow/borrower';
export * from './queries/borrow/collateralBorrower';
export * from './queries/borrow/market';
export * from './queries/collateral/types';
export * from './queries/collateral/whitelistCollateralQuery';
export * from './queries/collateral/whitelistCollateralByTokenAddrQuery';
export * from './queries/earn/apyHistory';
export * from './queries/earn/epochStates';
export * from './queries/gov/config';
export * from './queries/gov/distributionModelUpdateConfig';
export * from './queries/gov/myPolls';
export * from './queries/gov/poll';
export * from './queries/gov/polls';
export * from './queries/gov/state';
export * from './queries/gov/voters';
export * from './queries/market/anc';
export * from './queries/market/bAsset';
export * from './queries/market/buyback';
export * from './queries/market/collaterals';
export * from './queries/market/depositAndBorrow';
export * from './queries/market/stableCoin';
export * from './queries/market/state';
export * from './queries/market/ust';
export * from './queries/market/utils/dedupeTimestamp';
export * from './queries/mypage/txHistory';
export * from './queries/rewards/ancGovernanceRewards';
export * from './queries/rewards/ancUstLpRewards';
export * from './queries/rewards/claimableUstBorrowRewards';
export * from './queries/rewards/ustBorrowRewards';
export * from './tx/airdrop/claim';
export * from './tx/anc/ancUstLpProvide';
export * from './tx/anc/ancUstLpStake';
export * from './tx/anc/ancUstLpUnstake';
export * from './tx/anc/ancUstLpWithdraw';
export * from './tx/anc/buy';
export * from './tx/anc/governanceStake';
export * from './tx/anc/governanceUnstake';
export * from './tx/anc/sell';
export * from './tx/basset/claim';
export * from './tx/basset/export';
export * from './tx/basset/import';
export * from './tx/bond/burn';
export * from './tx/bond/claim';
export * from './tx/bond/mint';
export * from './tx/bond/swap';
export * from './tx/bond/withdraw';
export * from './tx/borrow/_fetchBorrowData';
export * from './tx/borrow/borrow';
export * from './tx/borrow/provideCollateral';
export * from './tx/borrow/redeemCollateral';
export * from './tx/borrow/repay';
export * from './tx/earn/deposit';
export * from './tx/earn/withdraw';
export * from './tx/gov/createPoll';
export * from './tx/gov/vote';
export * from './tx/rewards/allClaim';
export * from './tx/rewards/ancUstLpClaim';
export * from './tx/rewards/ustBorrowClaim';
export * from './tx/terra/send';
