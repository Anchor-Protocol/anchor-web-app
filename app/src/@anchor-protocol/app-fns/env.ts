import { Rate } from '@anchor-protocol/types';

export const ANCHOR_SAFE_RATIO: Rate<number> = 0.75 as Rate<number>;
export const ANCHOR_DANGER_RATIO: Rate<number> = 0.95 as Rate<number>;
