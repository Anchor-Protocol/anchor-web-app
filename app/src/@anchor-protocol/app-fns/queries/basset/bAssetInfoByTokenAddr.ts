import { CW20Addr, moneyMarket } from '@anchor-protocol/types';
import { QueryClient, wasmFetch, WasmQuery } from '@libs/query-client';
import { HumanAddr } from '@libs/types';
import { BAssetInfo, bAssetInfoQuery } from './bAssetInfo';

interface WhitelistWasmQuery {
  whitelist: WasmQuery<
    moneyMarket.overseer.Whitelist,
    moneyMarket.overseer.WhitelistResponse
  >;
}

type WhitelistElement = moneyMarket.overseer.WhitelistResponse['elems'][0];

async function bAssetInfoByTokenQuery(
  overseerContract: HumanAddr,
  queryClient: QueryClient,
  predicate: (elem: WhitelistElement) => boolean,
): Promise<WhitelistElement | undefined> {
  const { whitelist } = await wasmFetch<WhitelistWasmQuery>({
    ...queryClient,
    id: 'basset--list',
    wasmQuery: {
      whitelist: {
        contractAddress: overseerContract,
        query: {
          whitelist: {},
        },
      },
    },
  });
  return whitelist.elems.find(predicate);
}

export async function bAssetInfoByTokenAddrQuery(
  overseerContract: HumanAddr,
  tokenAddr: CW20Addr | undefined,
  queryClient: QueryClient,
): Promise<BAssetInfo | undefined> {
  const bAsset = await bAssetInfoByTokenQuery(
    overseerContract,
    queryClient,
    ({ collateral_token }) => tokenAddr === collateral_token,
  );

  if (!bAsset) {
    throw new Error(`Can't find bAssetInfo of "${tokenAddr}"`);
  }

  return bAssetInfoQuery(bAsset, queryClient);
}

export async function bAssetInfoByTokenSymbolQuery(
  overseerContract: HumanAddr,
  tokenSymbol: string | undefined,
  queryClient: QueryClient,
): Promise<BAssetInfo | undefined> {
  if (tokenSymbol === undefined) {
    return undefined;
  }

  const bAsset = await bAssetInfoByTokenQuery(
    overseerContract,
    queryClient,
    ({ symbol }) => symbol?.toLowerCase() === tokenSymbol?.toLowerCase(),
  );

  if (!bAsset) {
    throw new Error(`Can't find bAssetInfo of "${tokenSymbol}"`);
  }

  return bAssetInfoQuery(bAsset, queryClient);
}
