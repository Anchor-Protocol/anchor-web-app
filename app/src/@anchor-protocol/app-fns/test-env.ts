import { CW20Addr, HumanAddr } from '@anchor-protocol/types';

export const TEST_ADDRESSES = {
  bluna: {
    reward: 'terra1ac24j6pdxh53czqyrkr6ygphdeftg7u3958tl2' as HumanAddr,
    hub: 'terra1fflas6wv4snv8lsda9knvq2w0cyt493r8puh2e' as HumanAddr,
    airdropRegistry:
      'terra1u5ywhlve3wugzqslqvm8ks2j0nsvrqjx0mgxpk' as HumanAddr,
  },
  beth: {
    reward: 'terra1ja3snkedk4t0zp7z3ljd064hcln8dsv5x004na' as HumanAddr,
  },
  moneyMarket: {
    market: 'terra15dwd5mj8v59wpj0wvt233mf5efdff808c5tkal' as HumanAddr,
    collaterals: {
      bLuna: {
        type: 'bLuna',
        denom: 'ubluna',
        custody: 'terra1ltnkx0mv7lf2rca9f8w740ashu93ujughy4s7p' as HumanAddr,
        token: 'terra1u0t35drzyy0mujj8rkdyzhe264uls4ug3wdp3x' as CW20Addr,
      },
      bEth: {
        type: 'bEth',
        denom: 'ubeth',
        custody: 'terra1j6fey5tl70k9fvrv7mea7ahfr8u2yv7l23w5e6' as HumanAddr,
        token: 'terra19mkj9nec6e3y5754tlnuz4vem7lzh4n0lc2s3l' as CW20Addr,
      },
    },
    collateralsArray: [
      {
        type: 'bLuna',
        denom: 'ubluna',
        custody: 'terra1ltnkx0mv7lf2rca9f8w740ashu93ujughy4s7p' as HumanAddr,
        token: 'terra1u0t35drzyy0mujj8rkdyzhe264uls4ug3wdp3x' as CW20Addr,
      },
      {
        type: 'bEth',
        denom: 'ubeth',
        custody: 'terra1j6fey5tl70k9fvrv7mea7ahfr8u2yv7l23w5e6' as HumanAddr,
        token: 'terra19mkj9nec6e3y5754tlnuz4vem7lzh4n0lc2s3l' as CW20Addr,
      },
    ],
    overseer: 'terra1qljxd0y3j3gk97025qvl3lgq8ygup4gsksvaxv' as HumanAddr,
    oracle: 'terra1p4gg3p2ue6qy2qfuxtrmgv2ec3f4jmgqtazum8' as HumanAddr,
    interestModel: 'terra1m25aqupscdw2kw4tnq5ql6hexgr34mr76azh5x' as HumanAddr,
    distributionModel:
      'terra1u64cezah94sq3ye8y0ung28x3pxc37tv8fth7h' as HumanAddr,
  },
  liquidation: {
    liquidationContract:
      'terra16vc4v9hhntswzkuunqhncs9yy30mqql3gxlqfe' as HumanAddr,
  },
  anchorToken: {
    gov: 'terra16ckeuu7c6ggu52a8se005mg5c0kd2kmuun63cu' as HumanAddr,
    staking: 'terra19nxz35c8f7t3ghdxrxherym20tux8eccar0c3k' as HumanAddr,
    community: 'terra17g577z0pqt6tejhceh06y3lyeudfs3v90mzduy' as HumanAddr,
    distributor: 'terra1z7nxemcnm8kp7fs33cs7ge4wfuld307v80gypj' as HumanAddr,
    investorLock: 'not available in testnet' as HumanAddr,
    teamLock: 'not available in testnet' as HumanAddr,
    collector: 'terra1hlctcrrhcl2azxzcsns467le876cfuzam6jty4' as HumanAddr,
  },
  terraswap: {
    factory: 'terra18qpjm4zkvqnpjpw0zn0tdr8gdzvt8au35v45xf' as HumanAddr,
    blunaLunaPair: 'terra13e4jmcjnwrauvl2fnjdwex0exuzd8zrh5xk29v' as HumanAddr,
    ancUstPair: 'terra1wfvczps2865j0awnurk9m04u7wdmd6qv3fdnvz' as HumanAddr,
  },
  astroport: {
    generator: 'terra1gjm7d9nmewn27qzrvqyhda8zsfl40aya7tvaw5' as HumanAddr,
  },
  cw20: {
    bLuna: 'terra1u0t35drzyy0mujj8rkdyzhe264uls4ug3wdp3x' as CW20Addr,
    bEth: 'terra19mkj9nec6e3y5754tlnuz4vem7lzh4n0lc2s3l' as CW20Addr,
    aUST: 'terra1ajt556dpzvjwl0kl5tzku3fc3p3knkg9mkv8jl' as CW20Addr,
    ANC: 'terra1747mad58h0w4y589y3sk84r5efqdev9q4r02pc' as CW20Addr,
    AncUstLP: 'terra1agu2qllktlmf0jdkuhcheqtchnkppzrl4759y6' as CW20Addr,
    bLunaLunaLP: 'terra1tj4pavqjqjfm0wh73sh7yy9m4uq3m2cpmgva6n' as CW20Addr,
  },
};

export const TEST_WALLET_ADDRESS =
  'terra12hnhh5vtyg5juqnzm43970nh4fw42pt27nw9g9' as HumanAddr;
