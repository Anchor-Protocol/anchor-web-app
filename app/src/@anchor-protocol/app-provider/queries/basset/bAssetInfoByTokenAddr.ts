import { bAssetInfoByTokenAddrQuery } from '@anchor-protocol/app-fns';
import { CW20Addr } from '@anchor-protocol/types';
import { createQueryFn } from '@libs/react-query-utils';
import { useQuery, UseQueryResult } from 'react-query';
import { useAnchorWebapp } from '../../contexts/context';
import { ANCHOR_QUERY_KEY } from '../../env';
import { useQueryWithTokenDisplay } from '../utils/tokenDisplay';
import {
  BAssetInfoWithDisplay,
  withBAssetInfoTokenDisplay,
} from './utils/tokenDisplay';

const queryFn = createQueryFn(bAssetInfoByTokenAddrQuery);

export function useBAssetInfoByTokenAddrQuery(
  tokenAddr: CW20Addr | undefined,
): UseQueryResult<BAssetInfoWithDisplay | undefined> {
  const { queryClient, queryErrorReporter, contractAddress } =
    useAnchorWebapp();

  const bAssetInfo = useQuery(
    [
      ANCHOR_QUERY_KEY.BASSET_INFO_BY_TOKEN_ADDR,
      contractAddress.moneyMarket.overseer,
      tokenAddr,
      queryClient,
    ],
    queryFn,
    {
      refetchInterval: 1000 * 60 * 5,
      keepPreviousData: true,
      onError: queryErrorReporter,
    },
  );

  return useQueryWithTokenDisplay(bAssetInfo, withBAssetInfoTokenDisplay);
}
