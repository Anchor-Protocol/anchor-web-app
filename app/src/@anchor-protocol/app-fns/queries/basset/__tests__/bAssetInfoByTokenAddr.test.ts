import { TEST_ADDRESSES } from '@anchor-protocol/app-fns/test-env';
import { TEST_LCD_CLIENT } from '@libs/app-fns/test-env';
import { CW20Addr } from '@libs/types';
import { bAssetInfoByTokenAddrQuery } from '../bAssetInfoByTokenAddr';

describe('bAssetInfoByTokenAddrQuery', () => {
  test('should get result from query', async () => {
    const info = await bAssetInfoByTokenAddrQuery(
      TEST_ADDRESSES.moneyMarket.overseer,
      'terra19mkj9nec6e3y5754tlnuz4vem7lzh4n0lc2s3l' as CW20Addr,
      TEST_LCD_CLIENT,
    );

    expect(info).toMatchSnapshot();
  });
});
