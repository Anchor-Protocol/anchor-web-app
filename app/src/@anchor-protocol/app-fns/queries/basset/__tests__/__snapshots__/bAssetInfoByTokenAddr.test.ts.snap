// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bAssetInfoByTokenAddrQuery should get result from query 1`] = `
Object {
  "bAsset": Object {
    "collateral_token": "terra19mkj9nec6e3y5754tlnuz4vem7lzh4n0lc2s3l",
    "custody_contract": "terra1j6fey5tl70k9fvrv7mea7ahfr8u2yv7l23w5e6",
    "max_ltv": "0.6",
    "name": "Bonded ETH",
    "symbol": "BETH",
  },
  "converterConfig": Object {
    "anchor_token_address": "terra19mkj9nec6e3y5754tlnuz4vem7lzh4n0lc2s3l",
    "owner": "terra1pgsdhcc584ms5ncwp9qelyxycgaef5swt2fwue",
    "wormhole_token_address": "terra1d74gfj8gs6rskcuu80x3deus7gut77udhdajv7",
  },
  "custodyConfig": Object {
    "basset_info": Object {
      "decimals": 6,
      "name": "Bonded ETH",
      "symbol": "BETH",
    },
    "collateral_token": "terra19mkj9nec6e3y5754tlnuz4vem7lzh4n0lc2s3l",
    "liquidation_contract": "terra18j0wd0f62afcugw2rx5y8e6j5qjxd7d6qsc87r",
    "market_contract": "terra15dwd5mj8v59wpj0wvt233mf5efdff808c5tkal",
    "overseer_contract": "terra1qljxd0y3j3gk97025qvl3lgq8ygup4gsksvaxv",
    "owner": "terra16ckeuu7c6ggu52a8se005mg5c0kd2kmuun63cu",
    "reward_contract": "terra1ja3snkedk4t0zp7z3ljd064hcln8dsv5x004na",
    "stable_denom": "uusd",
  },
  "minter": Object {
    "cap": null,
    "minter": "terra1g68g7l3xkpm4hvadrqrfc53vtnfhl4dlnjm45u",
  },
  "wormholeTokenInfo": Object {
    "decimals": 8,
    "name": "bETH (Wormhole)",
    "symbol": "bETH",
    "total_supply": "10191140463",
  },
}
`;
